### PR TITLE
fix: add dnf_update to pre-install for Mariner to address kernel upgr…

### PR DESCRIPTION
**Pull request by SeanDougherty**
…ade issues

**What type of PR is this?**

**What this PR does / why we need it**:
Adding dnf_update to pre-install dependencies, which is before the reboot, so that the kernel is properly updated.

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
[Pipeline runs](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=73678641&view=results)
[alternate pipeline run](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=365325&view=results)

**Release note**:
```
none
```

### CODE REVIEW

**Review by hbeberman — APPROVED** [review]

File: vhdbuilder/packer/pre-install-dependencies.sh [hunk 1]
```diff
@@ -50,6 +50,9 @@ systemctlEnableAndStart sync-container-logs.service || exit 1
 
 # First handle Mariner + FIPS
 if [[ ${OS} == ${MARINER_OS_NAME} ]]; then
+  dnf_makecache || exit $ERR_APT_UPDATE_TIMEOUT
```
Comment (anujmaheshwari1) [inline vhdbuilder/packer/pre-install-dependencies.sh:53]: I wonder if this should be after the fips install is done?

Comment (SeanDougherty) [reply vhdbuilder/packer/pre-install-dependencies.sh:53]: + @dmcilvaney @rmhsawyer for their insight.

For context:
We are placing a call to dnf_makecache and dnf_update in pre-install-dependencies to trigger updating to the latest kernel before the reboot.

Question:
Should we wait until FIPS is enabled and the FIPS related grubby and dracut-fips packages are installed before updating?

Comment (SeanDougherty) [reply vhdbuilder/packer/pre-install-dependencies.sh:53]: @dmcilvaney or @rmhsawyer to comment

Is there a risk to installing an updated kernel before installingFIPS _and_ before rebooting into said newly updated kernel?

Comment (dmcilvaney) [reply vhdbuilder/packer/pre-install-dependencies.sh:53]: Generally all of Mariner's kernels and openssl are fips 'compliant' (certified means a specific version of the .rpm). So long as the grub cmdline is set correctly, and the dracut-fips package remains installed I don't see any issues with how we handle the kernel etc. packages.

**Review by dmcilvaney — APPROVED** [review]

**Review by anujmaheshwari1 — APPROVED** [review]
lgtm!

**Pull request merged**